src/capstone.rs

```diff
@@ -43,11 +43,17 @@ impl Instruction for capstone::Instr {
     }
 }
 
-impl Disassembler for capstone::Capstone {
-    fn architecture(&self) -> Architecture {
-        match disasembler.arch {
+impl From<capstone::cs_arch> for Architecture {
+    fn from(arch: capstone::cs_arch) -> Self {
+        match arch {
             cs_arch::CS_ARCH_X86 => X86,
             _ => panic!("Unexpected arch: {:?}", disassembler.arch),
```

Comment:
It seems more general to me to do a direct conversion from `cs_arch` to `Architecture`, instead of converting from `Disassembler` to `Architecture`, but I can change it.